Wikipedia/Code/ExploreCardViewController.swift

```diff
@@ -222,12 +222,12 @@ class ExploreCardViewController: PreviewingViewController, UICollectionViewDataS
         cell.selectionDelegate = self
     }
     
-    private func configureOnThisDayCell(_ cell: UICollectionViewCell, layoutOnly: Bool) {
-        guard let cell = cell as? OnThisDayExploreCollectionViewCell, let events = contentGroup?.contentPreview as? [WMFFeedOnThisDayEvent], events.count > 0 else {
+    private func configureOnThisDayCell(_ cell: UICollectionViewCell, forItemAt indexPath: IndexPath, layoutOnly: Bool) {
+        guard let cell = cell as? OnThisDayExploreCollectionViewCell, let events = contentGroup?.contentPreview as? [WMFFeedOnThisDayEvent], events.count > 0, events.indices.contains(indexPath.row) else {
             return
         }
-        let previousEvent: WMFFeedOnThisDayEvent? = events.count > 1 ? events[1] : events[0]
-        cell.configure(with: events[0], previousEvent: previousEvent, dataStore: dataStore, theme: theme, layoutOnly: layoutOnly)
+        let event = events[indexPath.row]
+        cell.configure(with: event, isFirst: events.first == event, isLast: events.last == event, dataStore: dataStore, theme: theme, layoutOnly: layoutOnly)
```

Comment:
this can be quicker index == 0, index == events.count - 1 checks instead of equality checks on the event objects

Reply:
or == events.indices.first, == events.indicies.last

Reply:
Fixed.